extensions/schema-compare/src/dialogs/schemaCompareOptionsDialog.ts

```diff
@@ -495,6 +500,7 @@ export class SchemaCompareOptionsDialog {
 				if (checkboxState && checkboxState.row !== undefined) {
 					let label = this.optionsLabels[checkboxState.row];
 					this.optionsLookup[label] = checkboxState.checked;
+					this.optionsChanged = true;
```

Comment:
Typically for this kind of popup you'd detect if there actually any changes. Right now if I check an option and then immediately uncheck it it'll see that as an option changing even though nothing has really changed.

It's not a huge deal but also isn't terribly hard to implement, just cache the state when the dialog is launched and then when OK is clicked compare to current state and pop up dialog if anything has changed.

Reply:
yeah it shouldn't be too hard to add, but for now this is the same behavior as SSDT